test/Dapr.AspNetCore.IntegrationTest/SubscribeEndpointTest.cs

```diff
@@ -107,7 +107,7 @@ public async Task SubscribeEndpoint_ReportsTopics()
                     subscriptions.Should().Contain(("pubsub", "metadata", "multiMetadataTopicAttr", string.Empty, string.Empty, "n1=v1;n2=v2,v3"));
                     subscriptions.Should().Contain(("pubsub", "metadata.1", "multiMetadataTopicAttr", "true", string.Empty, "n1=v1"));
                     subscriptions.Should().Contain(("pubsub", "splitMetadataTopicBuilder", "splitMetadataTopics", string.Empty, string.Empty, "n1=v1;n2=v1"));
-
+                    subscriptions.Should().Contain(("pubsub", "metadataseparator", "topicmetadataseparatorattr", string.Empty, string.Empty, "n1=v1|v2"));
```

Comment:
Can we also have a test for this using `WithTopic` just to make sure our different paths give us the same outcome?

Reply:
The design of "withtopic" does not require the use of separators at present